assets/js/blocks/podcast-platforms/edit.js

```diff
@@ -0,0 +1,237 @@
+import { useBlockProps, InspectorControls } from '@wordpress/block-editor';
+import { useState, useEffect } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+import { __ } from '@wordpress/i18n';
+import { useDebounce } from 'use-debounce';
+import {
+	Panel,
+	PanelBody,
+	PanelRow,
+	RangeControl,
+	SearchControl,
+	__experimentalItemGroup as ItemGroup,
+	__experimentalItem as Item,
+	BaseControl,
+	Button,
+	ButtonGroup,
+	Icon
+} from '@wordpress/components';
+
+
+function Edit( props ) {
+	const {
+		setAttributes,
+		isSelected,
+		attributes: {
+			showId,
+			iconSize,
+			align,
+		},
+	} = props;
+
+	/** State for the search text for the show name. Defaults to empty string. */
+	const [ searchText, setSearchText ] = useState( '' );
+
+	/** Debounced search text so that we don't trigger useEffect() for every character change. */
+	const [ debouncedSearchText ] = useDebounce( searchText, 300 );
+
+	/** State for search results matched by the search text. Defaults to array. */
+	const [ searchResults, setSearchResults ] = useState( [] );
+
+	/** State for the icon theme. Defaults to `color`. */
+	const [ iconTheme, setIconTheme ] = useState( 'color' );
+
+	/** State for platforms returned for a specific show. Defaults to array. */
+	const [ platforms, setPlatforms ] = useState( [] );
+
+	/**
+	 * Hits the `/wp/v2/search` endpoint to search for
+	 * podcast show by name.
+	 */
+	useEffect( () => {
+		const searchPodcastShow = async () => {
+			if ( ! searchText.length ) {
+				setSearchResults( [] );
+				return;
+			}
+
+			/** Query object required by `/wp/v2/search` to search for a term by name. */
+			const queryObject = {
+				search: searchText,
+				type: 'term',
+				subtype: 'podcasting_podcasts'
+			};
+
+			/** Converts an object to query-string. */
+			const queryString = new URLSearchParams( queryObject ).toString();
+
+			/** Returns the results of the search. */
+			const searchResults = await apiFetch( {
+				path: `/wp/v2/search?${ queryString }`,
+			} );
+
+			setSearchResults( searchResults );
+		};
+
+		searchPodcastShow();
+	}, [ debouncedSearchText ] );
+
+	/**
+	 * Fetches the podcasting platforms for a show whenever
+	 * showId updates.
+	 */
+	useEffect( () => {
+		if ( ! showId ) {
+			return;
+		}
+
+		/**
+		 * Responsible to fetch platforms for a show by show ID.
+		 * @returns void
+		 */
+		const fetchPlatforms = async () => {
+			const result = await apiFetch( {
+				url: `${ ajaxurl }?show_id=${ showId }&action=get_podcast_platforms`,
+			} );
+
+			if ( ! result.success ) {
+				return;
+			}
+
+			const {
+				data: { platforms, theme }
+			} = result;
+
+			setPlatforms( platforms );
+			setIconTheme( theme );
+		};
+
+		fetchPlatforms();
+	}, [ showId ] );
+
+	/**
+	 * Handler to set the attribute showId.
+	 *
+	 * @param {Int} termId The show ID.
+	 * @returns void
+	 */
+	const onShowSelect = ( termId ) => {
+		setAttributes( { showId: termId } );
```

Comment:
Maybe close the search list on select a Show and show a message if no podcast platforms are there for the selected show? For example: In the attached video, "TEST PODCAST" doesn't have any podcast platforms and selecting it not updates the UI and nothing happens on it. Maybe we should show a message like "No podcast platform message found". what do you think?

https://github.com/10up/simple-podcasting/assets/10613171/dafbac61-b1d6-4883-ad70-5420c8aab354